tinygrad/lazy.py

```diff
@@ -192,6 +192,7 @@ def toCPU(self):
   def cast(self:LazyBuffer, arg:DType) -> LazyBuffer: return elementwise_op(UnaryOps.CAST, self, arg=arg) if self.dtype != arg else self
   def unary_op(self:LazyBuffer, op:UnaryOps) -> LazyBuffer: return elementwise_op(op, self)
   def binary_op(self:LazyBuffer, op:BinaryOps, y:LazyBuffer) -> LazyBuffer: return elementwise_op(op, self, y)
+  def ternary_op(self:LazyBuffer, op:TernaryOps, y: LazyBuffer, z:LazyBuffer) -> LazyBuffer: return create_lazybuffer(self.device, ShapeTracker(self.shape), TernaryOps, LazyOp(op, (self, y, z,), None), y.dtype)
```

Comment:
Shouldn't this also call `elementwise_op` to allow for merged kernels?

Reply:
Oh yea, it has to merge.